src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -1645,5 +1645,173 @@ internal static Exception CreateInvalidNameCharException(string name, int index,
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(byte value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(ushort value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(uint value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(ulong value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(float value, Span<char> destination, out int charsWritten)
+        {
+            ReadOnlySpan<char> valueSpan = default;
+
+            if (float.IsNegativeInfinity(value))
+            {
+                valueSpan = "-INF";
+            }
+            else if (float.IsPositiveInfinity(value))
+            {
+                valueSpan = "INF";
+            }
+            else if (IsNegativeZero((double)value))
+            {
+                valueSpan = "-0";
+            }
+
+            if (valueSpan.IsEmpty)
+                return value.TryFormat(destination, out charsWritten, "R", NumberFormatInfo.InvariantInfo);
+
+            charsWritten = valueSpan.Length;
+            return valueSpan.TryCopyTo(destination);
+        }
+
+        internal static bool TryFormat(double value, Span<char> destination, out int charsWritten)
+        {
+            ReadOnlySpan<char> valueSpan = default;
+
+            if (double.IsNegativeInfinity(value))
+            {
+                valueSpan = "-INF";
+            }
+            else if (double.IsPositiveInfinity(value))
+            {
+                valueSpan = "INF";
+            }
+            else if (IsNegativeZero(value))
+            {
+                valueSpan = "-0";
+            }
+
+            if (valueSpan.IsEmpty)
+                return value.TryFormat(destination, out charsWritten, "R", NumberFormatInfo.InvariantInfo);
+
+            charsWritten = valueSpan.Length;
+            return valueSpan.TryCopyTo(destination);
+        }
+
+        internal static bool TryFormat(TimeSpan value, Span<char> destination, out int charsWritten)
+        {
+            return new XsdDuration(value).TryFormat(destination, out charsWritten);
+        }
+
+        internal static bool TryFormat(DateTime value, [StringSyntax(StringSyntaxAttribute.DateTimeFormat)] string format, Span<char> destination, out int charsWritten)
+        {
```

Comment:
I'm not familiar with this attribute beyond a quick search about it. Is it useful for an internal method? Does it propagate out the call stack in a useful way?

Reply:
I am not familiar with it too, just tried to follow syntax used in
public static string ToString(DateTime value, [StringSyntax(StringSyntaxAttribute.DateTimeFormat)] string format)

Maybe someone from the owners can help here with more information.

Reply:
@mconnew Check this https://github.com/dotnet/runtime/issues/62505